backend/src/services/auth/auth-login-service.ts

```diff
@@ -526,33 +536,85 @@ export const authLoginServiceFactory = ({
 
-    const selectedOrg = await orgDAL.findById(organizationId);
+    let rootOrg = selectedOrg;
+    let membershipRole;
+
+    if (isSubOrganization) {
+      if (!selectedOrg.rootOrgId) {
+        throw new BadRequestError({
+          message: "Invalid sub-organization"
+        });
+      }
+
+      rootOrg = await orgDAL.findById(selectedOrg.rootOrgId);
+      if (!rootOrg) {
+        throw new BadRequestError({
+          message: "Invalid root organization"
+        });
+      }
+
+      // Check user membership in the sub-organization
+      const orgMembership = await membershipUserDAL.findOne({
+        actorUserId: user.id,
+        scopeOrgId: organizationId,
+        scope: AccessScope.Organization,
+        status: OrgMembershipStatus.Accepted
+      });
+
+      if (!orgMembership) {
+        throw new ForbiddenRequestError({
+          message: `User does not have access to the sub-organization named ${selectedOrg.name}`
+        });
+      }
 
-    if (!selectedOrgMembership) {
-      throw new ForbiddenRequestError({
-        message: `User does not have access to the organization named ${selectedOrg?.name}`
+      // Check user membership in the root organization
+      const rootOrgMembership = await membershipUserDAL.findOne({
+        actorUserId: user.id,
+        scopeOrgId: rootOrg.id,
+        scope: AccessScope.Organization,
+        status: OrgMembershipStatus.Accepted
       });
+
+      if (!rootOrgMembership) {
+        throw new ForbiddenRequestError({
+          message: "User does not have access to the root organization"
+        });
+      }
+
+      membershipRole = (await membershipRoleDAL.findOne({ membershipId: orgMembership.id })).role;
+    } else {
+      // For root organizations, check membership using the existing method
+      const userOrgs = await orgDAL.findAllOrgsByUserId(user.id);
+      const selectedOrgMembership = userOrgs.find((org) => org.id === organizationId && org.userStatus !== "invited");
+
+      if (!selectedOrgMembership) {
+        throw new ForbiddenRequestError({
+          message: `User does not have access to the organization named ${selectedOrg.name}`
+        });
+      }
+      membershipRole = selectedOrgMembership.userRole;
     }
 
-    // Check if authEnforced is true and the current auth method is not an enforced method
     if (
       selectedOrg.authEnforced &&
       !isAuthMethodSaml(decodedToken.authMethod) &&
       decodedToken.authMethod !== AuthMethod.OIDC &&
-      !(selectedOrg.bypassOrgAuthEnabled && selectedOrgMembership.userRole === OrgMembershipRole.Admin)
+      !(selectedOrg.bypassOrgAuthEnabled && membershipRole === OrgMembershipRole.Admin)
     ) {
       throw new BadRequestError({
         message: "Login with the auth method required by your organization."
       });
     }
 
     if (selectedOrg.googleSsoAuthEnforced && decodedToken.authMethod !== AuthMethod.GOOGLE) {
```

Comment:
This needs to be based of root organization. Most of this data only exist in root organization

Reply:
The issue here is this means a user can directly login to sub-organization without doing the enforced google oauth